src/placeos-frontend-loader/api/remote/github.cr

```diff
@@ -0,0 +1,137 @@
+require "hash_file"
+require "./remote"
+require "octokit"
+
+module PlaceOS::FrontendLoader
+  class Github < Remote
+    def initialize(@metadata : Metadata = Metadata.instance)
+    end
+
+    private alias Remote = PlaceOS::FrontendLoader::Remote
+
+    @github_client = Octokit.client(GIT_USER, GIT_PASS)
+
+    # Returns the branches for a given repo
+    def branches(repo : String) : Hash(String, String)
+      uri = "https://api.github.com/repos/#{repo}/branches"
+      response = HTTP::Client.get uri
+      raise Exception.new("status_code for #{uri} was #{response.status_code}") unless (response.success? || response.status_code == 302)
+      branches = Hash(String, String).new
+      Array(JSON::Any).from_json(response.body).map do |value|
+        branch_name = value["name"].to_s.strip.lchop("origin/")
+        branches[branch_name] = value["commit"]["sha"].to_s
+      end
+      branches
```

Comment:
This pattern ain't quite right.
`.map` is constructing an array that you're not using.

One way of going about this would be...
```suggestion
      Array(JSON::Any).from_json(response.body).map do |value|
        branch_name = value["name"].to_s.strip.lchop("origin/")
        ({branch_name, value["commit"]["sha"].to_s})
      end.to_h
```

the other being...
```suggestion
      Array(JSON::Any)
        .from_json(response.body)
        .each_with_object(Hash(String, String).new) do |value, branches|  
          branch_name = value["name"].to_s.strip.lchop("origin/")
          branches[branch_name] = value["commit"]["sha"].to_s
        end
```

Reply:
You've also gone back to using `JSON::Any` instead of objects.
In the interest of moving on, it's not that big of an issue as this isn't a particularly load heavy service.

Reply:
Yeah these functions aren't going to be used anymore anyways cause github/gitlab will all just use git cli